### PR TITLE
Modify viscosity in stabilization tau

### PR DESCRIPTION
**Pull request by hepap**
# Description of the problem

- For phase change problem, the viscosity used in the tau computation for the stabilization in the solid used the numerical solid viscosity. Since this solid viscosity has no physical meaning, it was weird to used it, and the linear solver seemed to converge slower.

# Description of the solution

- Introduce new attributes in NS scratch data `kinematic_viscosity_for_stabilization` and `dynamic_viscosity_for_stabilization`. Both are computed by the rheology model. For now, except in the phase change, it returns the current viscosity. However, in the phase change rheology model, it returns the liquid viscosity.

# How Has This Been Tested?

- The melting cavity case was used to insure that the results stay the same and to show better convergence of the linear solver.

- [x] heat_transfer_phase_change_constrain_solid_domain
- [x] heat_transfer_vof_phase_change_constrain_solid_domain

Those two tests were changed to have a more representative behaviour.

# Future changes

- Test the viscosity scale in VOF and non-newtonian cases to see if it brings some gains

### CODE REVIEW

**Review by PierreLaurentinCS — COMMENTED** [review]
Very nice additions, I left some questions here and there but overall it's a great work!

**Review by PierreLaurentinCS — CHANGES_REQUESTED** [review]
See my previous comments for requested changes, then it'll be good to go :)

File: applications_tests/lethe-fluid/heat_transfer_phase_change_constrain_solid_domain.output [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
 Pressure drop: 0 Pa
 Total pressure drop: 0 Pa
 
 *******************************************************************************
-Transient iteration: 1        Time: 0.1      Time step: 0.1      CFL: 0       
+Transient iteration: 1        Time: 0.01     Time step: 0.01     CFL: 0       
 *******************************************************************************
-Pressure drop: -0.42844 Pa
-Total pressure drop: -0.42844 Pa
+Pressure drop: -1.51849e-18 Pa
```
[the remark this answers is not in the document]
Comment (hepap) [reply applications_tests/lethe-fluid/heat_transfer_phase_change_constrain_solid_domain.output:12]: My guess would be that in the first time step (smaller than it use to be), there is not much melted fluid, so no motion and no pressure drop. What do you think @blaisb?

File: include/core/rheological_model.h [hunk 2]
```diff
@@ -703,6 +776,15 @@ class PhaseChangeRheology : public RheologicalModel
     const field /*id*/,
     std::vector<double> &jacobian_vector) override;
 
+  /**
+   * @brief Returns the kinematic viscosity scale.
+   */
+  double
+  get_kinematic_viscosity_scale() const override
+  {
+    return param.kinematic_viscosity_l;
```
[the remark this answers is not in the document]
Comment (PierreLaurentinCS) [reply include/core/rheological_model.h:785]: This makes much more sense, thanks

**Review by PierreLaurentinCS — APPROVED** [review]
Good!

**Review by blaisb — APPROVED** [review]

**Comment by blaisb** [comment]
@hepap can you rebase? This will then be ready for merge

**Comment by hepap** [comment]
@blaisb it's rebased!

**Comment by blaisb** [comment]
@hepap something went wrong in your rebase I think

**Pull request merged**